src/fides/api/service/messaging/messaging_crud_service.py

```diff
@@ -80,32 +88,283 @@ def get_messaging_config_by_key(db: Session, key: FidesKey) -> MessagingConfigRe
+    """
+    Retrieve all templates from the database, filling in default values if any default template type
+    is not found in the database.
+    """
+    #
     templates_from_db = {
-        template.key: template.content for template in MessagingTemplate.all(db)
+        template.type: template.content for template in MessagingTemplate.all(db)
     }
-
-    # Create a list of MessagingTemplate models, using defaults if a key is not found in the database
     templates = []
-    for key, template in DEFAULT_MESSAGING_TEMPLATES.items():
-        content = templates_from_db.get(key, template["content"])
+    for template_type, template in DEFAULT_MESSAGING_TEMPLATES.items():
+        content = templates_from_db.get(template_type, template["content"])
         templates.append(
             MessagingTemplate(
-                key=key,
+                type=template_type,
                 content=content,
             )
         )
 
     return templates
 
 
-def get_messaging_template_by_key(db: Session, key: str) -> Optional[MessagingTemplate]:
-    template = MessagingTemplate.get_by(db, field="key", value=key)
+def get_basic_messaging_template_by_type(
+    db: Session, template_type: str
+) -> Optional[MessagingTemplate]:
+    template = MessagingTemplate.get_by(db, field="type", value=template_type)
 
     # If no template is found in the database, use the default
-    if not template and key in DEFAULT_MESSAGING_TEMPLATES:
-        content = DEFAULT_MESSAGING_TEMPLATES[key]["content"]
-        template = MessagingTemplate(key=key, content=content)
+    if not template and template_type in DEFAULT_MESSAGING_TEMPLATES:
+        content = DEFAULT_MESSAGING_TEMPLATES[template_type]["content"]
+        template = MessagingTemplate(type=template_type, content=content)
+
+    return template
+
+
+def create_or_update_basic_templates(
+    db: Session, data: Dict[str, Any]
+) -> Optional[MessagingTemplate]:
+    """
+    For "basic", or non property-specific messaging templates, we update if template "type" matches an existing db row,
+    otherwise we create a new one.
 
+    There might be multiple templates configured by type, in the edge case where a paid user downgrades to OSS.
+    We use the one associated with the default property if found. If no default, we fall back on first item for safety.
+    """
+    template = None
+    templates = (
+        MessagingTemplate.query(db=db)
+        .filter(MessagingTemplate.type == data["type"])
+        .all()
+    )
+
+    if len(templates) > 1:
+        default_property = Property.get_by(db=db, field="is_default", value=True)
+        if default_property:
+            template = MessagingTemplate.filter(
+                db=db,
+                conditions=(
+                    (MessagingTemplate.type == data["type"])
+                    & (Property.id == default_property.id)
+                ),
+            ).first()
+    elif len(templates) == 1:
+        template = templates[0]
```

Comment:
For readability, I also might swap these conditional, put the 1 case first, and the more than 1 second, just my opinion

Reply:
Agreed, thanks!